be/src/formats/parquet/file_reader.cpp

```diff
@@ -556,17 +571,17 @@ Status FileReader::_init_group_readers() {
 
 Status FileReader::_prepare_cur_row_group() {
     auto& r = _row_group_readers[_cur_row_group_idx];
+    // prepare row group
+    RETURN_IF_ERROR(r->prepare());
     // if coalesce read enabled, we have to
     // 0. clear last group memory
     // 1. allocate shared buffered input stream and
     // 2. collect io ranges of every row group reader.
     // 3. set io ranges to the stream.
     if (config::parquet_coalesce_read_enable && _sb_stream != nullptr) {
-        // clear last group memory;
-        _sb_stream->release();
```

Comment:
why remove it?  could not we release it?

Reply:
the memory of current row group's data range will be released when it is closing, and we can't release all shared buffer because pageindex of all row group in it.

Reply:
w'd better to separate them. because if we process two row groups, we expect to release first row group memory when we handle the second one. Otherwise memory usage will be doubled.

Reply:
we actually will release the first row group's memory when we handle the second one, the first row group's memory will be released when the first row group is closing. we just hold the pageindex of all row group which always is small.